internal/providers/terraform/aws/db_instance.go

```diff
@@ -192,6 +191,36 @@ func NewDBInstance(d *schema.ResourceData, u *schema.UsageData) *schema.Resource
 		})
 	}
 
+	if !d.IsEmpty("backup_retention_period") || (u != nil && !u.IsEmpty("additional_backup_storage_gb")) {
+		var backupStorageGB *decimal.Decimal
+		if u != nil && !u.IsEmpty("additional_backup_storage_gb") {
+			backupStorageGB = decimalPtr(decimal.NewFromInt(u.Get("additional_backup_storage_gb").Int()))
+		}
+
+		backupStorageDbEngine := "Any"
+		if strings.HasPrefix(*databaseEngine, "Aurora") {
+			backupStorageDbEngine = *databaseEngine
+		}
+
+		costComponents = append(costComponents, &schema.CostComponent{
+			Name:            "Additional backup storage",
+			Unit:            "GB",
+			UnitMultiplier:  decimal.NewFromInt(1),
+			MonthlyQuantity: backupStorageGB,
+			ProductFilter: &schema.ProductFilter{
+				VendorName:    strPtr("aws"),
+				Region:        strPtr(region),
+				Service:       strPtr("AmazonRDS"),
+				ProductFamily: strPtr("Storage Snapshot"),
+				AttributeFilters: []*schema.AttributeFilter{
+					{Key: "databaseEngine", Value: strPtr(backupStorageDbEngine)},
+					{Key: "usagetype", ValueRegex: strPtr("/BackupUsage/i")},
+					{Key: "engineCode", ValueRegex: strPtr("/[0-9]+/")},
```

Comment:
Do we need an `engineCode` filter? Looks like it's always a number anyway.

Reply:
So this is a weird one, and I am very much so open to suggestions because I don't like having to do it either... but since you asked for it... get ready (wall of text). 

I can instead use the following `usagetype` query after playing around with it more: `"/(RDS:|Aurora:)(Charged)?BackupUsage/i"`, do you think that would be more elegant? For a longer explanation as for why I initially included `engineCode`: 

For every `Any` databaseEngine, there exists a "non-custom" version of the database in the `cloud-pricing-api` database, and a "custom" version of the database (see: https://aws.amazon.com/rds/custom/). I ran the `cloud-pricing-api` database tool locally to confirm my queries / attribute filters would only return _one_ result during any construction of a `aws_db_instance`. 

Scenario: Calculating the costs of a `aws_db_instance` with a `backupStorageDbEngine` value set to `Any` in the region `us-east-1`.
The following psql query will represent this search in the `cloud-pricing-api` **if I exclude the `AttributeFilter` for  `engineCode`** (only selecting the `attributes` and `prices` columns here since that is most relevant):
```psql
SELECT attributes FROM products WHERE "vendorName" = 'aws' AND "service" = 'AmazonRDS' AND "productFamily" = 'Storage Snapshot' AND "attributes"->>'databaseEngine' = 'Any' AND "attributes"->>'usagetype' LIKE '%BackupUsage%' AND "region" = 'us-east-1';
```

However, here are the results:
```
                                                                                                                                                          attributes                                                                       |                             prices
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"location": "US East (N. Virginia)", "operation": "", "usagetype": "RDSCustom:ChargedBackupUsage", "regionCode": "us-east-1", "servicecode": "AmazonRDS", "servicename": "Amazon Relational Database Service", "locationType": "AWS Region", "databaseEngine": "Any", "databaseEdition": "Any", "deploymentModel": "Custom"} | {"70bc2b1d05db0a22bbad7b1a9b539b70-ee3dd7e4624338037ca6fea0933a662f": [{"USD": "0.0950000000", "unit": "GB-Mo", "priceHash": "70bc2b1d05db0a22bbad7b1a9b539b70-ee3dd7e462433803
7ca6fea0933a662f", "description": "USD 0.095 per GB-Month of snapshot storage", "endUsageAmount": "Inf", "purchaseOption": "on_demand", "startUsageAmount": "0", "effectiveDateStart": "2021-11-01T00:00:00Z"}]}
 {"location": "US East (N. Virginia)", "operation": "", "usagetype": "RDS:ChargedBackupUsage", "engineCode": "0", "regionCode": "us-east-1", "servicecode": "AmazonRDS", "servicename": "Amazon Relational Database Service", "locationType": "AWS Region", "storageMedia": "AmazonS3", "databaseEngine": "Any"}               | {"87eb8774ee46ef6838ab3918adb95698-ee3dd7e4624338037ca6fea0933a662f": [{"USD": "0.0950000000", "unit": "GB-Mo", "priceHash": "87eb8774ee46ef6838ab3918adb95698-ee3dd7e4624338037ca6fea0933a662f", "description": "$0.095 per additional GB-month of backup storage exceeding free allocation", "endUsageAmount": "Inf", "purchaseOption": "on_demand", "startUsageAmount": "0", "effectiveDateStart": "2021-11-01T00:00:00Z"}]}
(2 rows)
```

As you can see there are two results returned, if we look strictly at the attributes column, and the values in that row, we can see that the difference between the two is that one is designated as a `Custom` RDS database solution (https://aws.amazon.com/rds/custom/). [Terraform ](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/db_instance)has no support for "Custom" RDS databases as far as I can tell by their documentation. 

When I inspected the two rows (one custom, one non-custom), there were very few differences I could find.
```
 {"location": "US East (N. Virginia)", "operation": "", "usagetype": "RDSCustom:ChargedBackupUsage", "regionCode": "us-east-1", "servicecode": "AmazonRDS", "servicename": "Amazon Relational Database Service", "locationType": "AWS Region", "databaseEngine": "Any", "databaseEdition": "Any", "deploymentModel": "Custom"}
 {"location": "US East (N. Virginia)", "operation": "", "usagetype": "RDS:ChargedBackupUsage", "engineCode": "0", "regionCode": "us-east-1", "servicecode": "AmazonRDS", "servicename": "Amazon Relational Database Service", "locationType": "AWS Region", "storageMedia": "AmazonS3", "databaseEngine": "Any"}
```

Namely: 
- `usagetype` differs (however, `usagetype` also is different for all four database types for a particular region, `Aurora:BackupUsage`, `RDSCustom:ChargedBackupUsage`, `AuroraBackupUsage` and `RDS:ChargedBackupUsage`)
  - I could update the Regex filter here using the following regex filter: `"/(RDS:|Aurora:)(Charged)?BackupUsage/i"` which would obviate the need for the `engineCode` filter. 
- All databases _except_ the "Custom" RDS database included the `engineCode` attribute. By filtering for this attribute, it would exclude the "Custom" RDS database pricing. 

(for reference, here are all the database pricing attributes for backup storage in `us-east-1`):
```
cloud_pricing=# SELECT attributes FROM products WHERE "vendorName" = 'aws' AND "service" = 'AmazonRDS' AND "productFamily" = 'Storage Snapshot' AND "attributes"->>'databaseEngine' IS NOT NULL AND "attributes"->>'usagetype' LIKE '%BackupUsage%' AND"region" = 'us-east-1';
                                                                                                                                                                   attributes
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 {"location": "US East (N. Virginia)", "operation": "CreateDBInstance:0016", "usagetype": "Aurora:BackupUsage", "engineCode": "16", "regionCode": "us-east-1", "servicecode": "AmazonRDS", "servicename": "Amazon Relational Database Service", "locationType": "AWS Region", "storageMedia": "AmazonS3", "databaseEngine": "Aurora MySQL"}
 {"location": "US East (N. Virginia)", "operation": "", "usagetype": "RDSCustom:ChargedBackupUsage", "regionCode": "us-east-1", "servicecode": "AmazonRDS", "servicename": "Amazon Relational Database Service", "locationType": "AWS Region", "databaseEngine": "Any", "databaseEdition": "Any", "deploymentModel": "Custom"}
 {"location": "US East (N. Virginia)", "operation": "CreateDBInstance:0021", "usagetype": "Aurora:BackupUsage", "engineCode": "21", "regionCode": "us-east-1", "servicecode": "AmazonRDS", "servicename": "Amazon Relational Database Service", "locationType": "AWS Region", "storageMedia": "AmazonS3", "databaseEngine": "Aurora PostgreSQL"}
 {"location": "US East (N. Virginia)", "operation": "", "usagetype": "RDS:ChargedBackupUsage", "engineCode": "0", "regionCode": "us-east-1", "servicecode": "AmazonRDS", "servicename": "Amazon Relational Database Service", "locationType": "AWS Region", "storageMedia": "AmazonS3", "databaseEngine": "Any"}
```

Reply:
Thanks @joshniec, that was a great explanation 🙌 . This makes me wonder in the future if we'll need better filtering (!=, not contains, etc). I think the way you have it makes sense just now, we might just want to add a comment to say why we're filtering on `engineCode`.